app/controllers/OrderController.java

```diff
@@ -3,58 +3,67 @@
     }
 
-    public Result show() {
-        final Cart currentCart = cartService.getOrCreateCart(session());
-        if (!currentCart.getLineItems().isEmpty()) {
-            final int selectedFrequency = cartService.getFrequency(currentCart.getId());
-            if (selectedFrequency > 0) {
-                final ProductVariant selectedVariant = currentCart.getLineItems().get(0).getVariant();
-                final OrderPageData orderPageData = new OrderPageData(selectedVariant, selectedFrequency, currentCart);
-                return ok(order.render(orderPageData));
-            } else {
-                flash("error", "Missing frequency of delivery. Please try selecting it again.");
+    public F.Promise<Result> show() {
+        LOG.debug("Display Order details page");
+        final F.Promise<Cart> currentCartPromise = cartService.getOrCreateCart(session());
+        return currentCartPromise.flatMap(currentCart -> {
+            if (!currentCart.getLineItems().isEmpty()) {
+                //TODO
+                //sometimes this seems to be 0 and the flash message occurs, sometimes not
+                final F.Promise<Integer> selectedFrequencyPromise = cartService.getFrequency(currentCart.getId());
+                final F.Promise<Result> resultPromise = selectedFrequencyPromise.map(selectedFrequency -> {
+                    if (selectedFrequency > 0) {
+                        final ProductVariant selectedVariant = currentCart.getLineItems().get(0).getVariant();
+                        final OrderPageData orderPageData = new OrderPageData(selectedVariant, selectedFrequency, currentCart);
+                        return ok(order.render(orderPageData));
+                    } else {
+                        flash("error", "Missing frequency of delivery. Please try selecting it again.");
+                        return redirect(routes.ProductController.show());
+                    }
+                });
+                return resultPromise;
             }
-        }
-        return redirect(routes.ProductController.show());
+            flash("error", "Please select a box and how often you want it.");
+            return F.Promise.pure(redirect(routes.ProductController.show()));
+        });
     }
 
-    public Result submit() {
-        try {
-            final Cart currentCart = cartService.getOrCreateCart(session());
-            final Cart clearedCart = cartService.clearCart(currentCart);
-            return ok(success.render());
-        } catch (Exception e) {
-            Logger.error(e.getMessage(), e);
-            return internalServerError();
-        }
+    //TODO check is doing nothing than clear the cart?!
+    public F.Promise<Result> submit() {
+        LOG.debug("Submitting Order details page");
+        final F.Promise<Cart> currentCartPromise = cartService.getOrCreateCart(session());
+        return currentCartPromise.map(currentCart -> ok(success.render()));
```

Comment:
In here you will eventually have to clear the session, once https://github.com/commercetools/sphere-donut/issues/29 is implemented. Also maybe the customer according to https://github.com/commercetools/sphere-donut/issues/30